File: PROGRAMMERS/소수만들기.py
# nums = [1,2,3,4]
nums = [1,2,7,6,4]

def is_prime(x):
    if x < 2:
        return False
    i = 2
    while i * i <= x:
        if x % i == 0:
            return False
        i += 1
    return True

def solution(nums):
    answer = -1
    cnt = 0
    for i in range(len(nums)):
        for j in range(i+1, len(nums)):
            num = 0
            for k in range(j+1, len(nums)):
                num = nums[i] + nums[j] + nums[k]
                if is_prime(num):
                    cnt += 1

    answer = cnt
    return answer

print(solution(nums))

File: PROGRAMMERS/JAVA/src/Lv_1/소수만들기.java
package Lv_1;

import java.util.ArrayList;
import java.util.List;

public class 소수만들기 {
    public int is_prime(int num) {
        if (num == 0 || num == 1) {
            return 0;
        }
        if (num == 2) {
            return 1;
        }
        for (int i = 2; i <= (int) Math.sqrt(num); i++) {
            if (num % i == 0) {
                return 0;
            }
        }
        return 1;
    }
    public int solution(int[] nums) {
        int answer = 0;
        List<Integer> numbers = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                for (int k = j + 1; k < nums.length; k++) {
                    numbers.add(nums[i] + nums[j] + nums[k]);
                }
            }
        }
        for (int i = 0; i < numbers.size(); i++) {
            if (is_prime(numbers.get(i)) == 1) {
                answer++;
            }
        }
        return answer;
    }
}
